data/lua/wml/random_placement.lua

```diff
@@ -32,6 +32,7 @@ wesnoth.wml_actions.random_placement = function(cfg)
 		wesnoth.set_variable(variable .. ".x", point[1])
 		wesnoth.set_variable(variable .. ".y", point[2])
 		wesnoth.set_variable(variable .. ".n", i)
+		wesnoth.set_variable(variable .. ".terrain", wesnoth.get_terrain(point[1], point[2]))
```

Comment:
wesnoth.get_terrain returns a string with of the terrain code. liek th terrain= yes in store_locations